qiskit_ibm_provider/ibm_backend.py

```diff
@@ -457,90 +456,117 @@ def run(
                 - If ESP readout is used and the backend does not support this.
         """
         # pylint: disable=arguments-differ
+
         validate_job_tags(job_tags, IBMBackendValueError)
 
         status = self.status()
         if status.operational is True and status.status_msg != "active":
             warnings.warn(f"The backend {self.name} is currently paused.")
 
-        sim_method = None
-        if self.configuration().simulator:
-            sim_method = getattr(self.configuration(), "simulation_method", None)
-        if noise_model:
-            try:
-                noise_model = noise_model.to_dict()
-            except AttributeError:
-                pass
-        measure_esp_enabled = getattr(
-            self.configuration(), "measure_esp_enabled", False
-        )
-        # set ``use_measure_esp`` to backend value if not set by user
-        if use_measure_esp is None:
-            use_measure_esp = measure_esp_enabled
-        if use_measure_esp and not measure_esp_enabled:
-            raise IBMBackendValueError(
-                "ESP readout not supported on this device. Please make sure the flag "
-                "'use_measure_esp' is unset or set to 'False'."
-            )
+        program_id = "circuit-runner"
+        if dynamic or isinstance(circuits, str):
+            # str circuit means QASM3 string
+            program_id = "qasm3-runner"
 
         if isinstance(shots, float):
             shots = int(shots)
-
         if not self.configuration().simulator:
             circuits = self._deprecate_id_instruction(circuits)
+        inputs = {"circuits": circuits}
+        options = {"backend": self.name}
+
+        if job_name:
+            inputs["job_name"] = job_name
+        if max_circuits_per_job:
+            inputs["max_circuits_per_job"] = max_circuits_per_job
+        if header:
+            inputs["header"] = header
+        if shots:
+            inputs["shots"] = shots
+        if memory is not None:
+            inputs["memory"] = memory
```

Comment:
Separate class seems like a good idea; I'll need to know which options are not supported anymore.

I don't use `_get_run_config` since it was not used in the runtime run method, which I base this method on. I'll try to see how to integrate it here.

Reply:
> I'll need to know which options are not supported anymore.

@taalexander can give you a definitive answer on what options are supported. Note that the QASM2 path still supports all of the options, just not QASM3.